akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -546,8 +537,13 @@ private[akka] class PersistentShard(
       log.warning("PersistentShard snapshot failure: {}", reason.getMessage)
 
     case DeleteMessagesSuccess(toSequenceNr) ⇒
-      log.debug("PersistentShard messages to {} deleted successfully", toSequenceNr)
-      deleteSnapshots(SnapshotSelectionCriteria(maxSequenceNr = toSequenceNr - 1))
+      val deleteFrom = math.max(0, toSequenceNr - (keepNrOfBatches * snapshotAfter))
```

Comment:
shouldn't this be `deleteTo  - (keepNrOfBatches * snapshotAfter)`, e.g. otherwise would be wrong when keepNrOfBatches is 0

Reply:
Yes, good catch